presto-kafka/src/main/java/io/prestosql/plugin/kafka/KafkaPageSink.java

```diff
@@ -59,17 +61,20 @@ public KafkaPageSink(
         this.messageEncoder = requireNonNull(messageEncoder, "messageEncoder is null");
         requireNonNull(producerFactory, "producerFactory is null");
         this.producer = producerFactory.create();
-        this.errorCounter = new ErrorCountingCallback();
+        this.producerCallback = new ProducerCallback();
+        this.expectedWrittenBytes = 0;
     }
 
-    private static class ErrorCountingCallback
+    private static class ProducerCallback
             implements Callback
     {
         private final AtomicLong errorCounter;
```

Comment:
We originally used `AtomicLong` for `errorCounter` but it looks like the callback is not called concurrently from multiple threads. 
Please add a commit which changes `errorCounter` to just `long`, like you did with `writtenBytes`.